adb_client.c

```diff
@@ -129,24 +129,17 @@ static void send_auth_request(adb_client_t *client, apacket *p)
 #endif /* CONFIG_ADBD_AUTHENTICATION */
 
 static void handle_open_frame(adb_client_t *client, apacket *p) {
-    adb_service_t *svc;
+    adb_service_t *svc = NULL;
     char *name = (char*) p->data;
 
     /* OPEN(local-id, 0, "destination") */
-    if (p->msg.arg0 == 0 || p->msg.arg1 != 0) {
-    	return;
+    if (p->msg.arg0 != 0 && p->msg.arg1 == 0) {
+        name[p->msg.data_length > 0 ? p->msg.data_length - 1 : 0] = 0;
+        svc = adb_service_open(client, name, p);
     }
 
-    name[p->msg.data_length > 0 ? p->msg.data_length - 1 : 0] = 0;
-    svc = adb_service_open(client, name, p);
     if(svc == NULL) {
-        if (p->msg.arg1 != 0) {
```

Comment:
This is actually required, although it's a little bit hackish and not documented.

In the current implementation a service can in `adb_service_open()` update `p->msg.arg1` and return `NULL`. This means that the service fully ran during the `adb_service_open()` and wants to send data back to ADB client with `OKAY` frame.

Some commands need this logic.

Reply:
Thanks for explanation, the code skip to change this case.